lib/large_object_store.rb

```diff
@@ -53,6 +55,11 @@ def read(key)
         return nil if slices.compact.size < pages
         slices.join("")
       end
+
+      if data.getbyte(0) == 0x78 && [0x01,0x9C,0xDA].include?(data.getbyte(1))
```

Comment:
Oops, merged too fast? Do you want to do that? I can merge the next commit and then release